apps/ui/src/routes/index.ts

```diff
@@ -32,4 +33,34 @@ const router = createRouter({
   }
 });
 
+// Add a global navigation guard for URL redirection
+router.beforeEach((to, _from, next) => {
+  let redirectPath: string | null = null;
+
+  // Match and redirect paths like "/safe.eth/settings" to "/s:safe.eth/settings"
+  const domainMatch = to.path.match(/^\/([^:\/]+?\.[^:\/]+)(\/.*)?$/);
+  if (domainMatch) {
+    const domain = domainMatch[1];
+    const rest = domainMatch[2] || '';
+    redirectPath = `/${metadataNetwork}:${domain}`;
+    if (rest && !/^\/about$/.test(rest)) {
+      redirectPath += rest;
+    }
+  }
+
+  // Match and redirect paths like "/delegate/safe.eth" to "/s:safe.eth/delegates"
+  const delegateMatch = to.path.match(/^\/delegate\/([^:\/]+?\.[^:\/]+)$/);
+  if (delegateMatch) {
+    const domain = delegateMatch[1];
+    redirectPath = `/${metadataNetwork}:${domain}/delegates`;
```

Comment:
This will not work for testnet

Reply:
Why not?

Reply:
Hum, we don't support delegates API on testnet?

Reply:
Ah no, apologies, was thinking of something else